src/core/border_router/routing_manager.cpp

```diff
@@ -603,47 +684,75 @@ void RoutingManager::EvaluateNat64Prefix(void)
 {
     OT_ASSERT(mIsRunning);
 
-    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
-    NetworkData::ExternalRouteConfig config;
-    Ip6::Prefix                      smallestNat64Prefix;
+    Error                            error;
+    NetworkData::ExternalRouteConfig preferredNat64PrefixConfig;
 
     LogInfo("Evaluating NAT64 prefix");
 
-    smallestNat64Prefix.Clear();
-    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, config) == kErrorNone)
+    error = Get<NetworkData::Leader>().GetPreferredNat64Prefix(preferredNat64PrefixConfig);
+
+    // (Re)advertise NAT64 prefix from this BR when
```

Comment:
Do we need to advertise infra NAT64 prefix if a low priority prefix is found in network data?

Reply:
No if the low priority prefix is advertised by another BR. `The Border Router MUST advertise its NAT64 prefix in the Thread Network Data if no other Border Router is already advertising a NAT64 prefix.`

Reply:
Section 9.7.3.3 seems to indicate that the larger prefix should be withdrawn despite of its preference.
Should a medium preference prefix be preferred than a low preference prefix?

@xuyirio @jwhui @EskoDijk

Reply:
@simonlingoogle The intention is for the higher-priority prefix to always win over a lower-priority prefix. So an infrastructure-obtained prefix at medium priority will win over a self-generated prefix at low priority. Seems the spec forgets to mention this!

Reply:
Agreed. Would be good to provide this feedback to TC. We had the same discussion and outcome for OMR prefix. Some code might be reused from there?

Reply:
Thanks for the clarification! Updated the logic to allow a BR to add its infrastructure prefix even if there is an existing low preference prefix.